code/modules/hydroponics/beekeeping/beehive.dm

```diff
@@ -1,308 +1,455 @@
-		if(smoked)
+		if(time_end_smoked >= REALTIMEOFDAY)
 			to_chat(user, "The hive is smoked.")
-		return 1
-	else if(IS_SCREWDRIVER(I))
-		if(bee_count)
-			to_chat(user, "<span class='notice'>You can't dismantle \the [src] with these bees inside.</span>")
-			return
-		to_chat(user, "<span class='notice'>You start dismantling \the [src]...</span>")
-		playsound(loc, 'sound/items/Screwdriver.ogg', 50, 1)
-		if(do_after(user, 30, src))
-			user.visible_message("<span class='notice'>\The [user] dismantles \the [src].</span>", "<span class='notice'>You dismantle \the [src].</span>")
-			new /obj/item/beehive_assembly(loc)
-			qdel(src)
-		return
+		return TRUE
 
-/obj/machinery/beehive/physical_attack_hand(var/mob/user)
-	if(!closed)
-		. = TRUE
-		if(honeycombs < 100)
-			to_chat(user, "<span class='notice'>There are no filled honeycombs.</span>")
-			return
-		if(!smoked && bee_count)
-			to_chat(user, "<span class='notice'>The bees won't let you take the honeycombs out like this, smoke them first.</span>")
+	else if(istype(I, /obj/item/grab))
+		var/obj/item/grab/G = I
+		var/mob/living/GM   = G.get_affecting_mob()
+		if(!GM)
 			return
-		user.visible_message("<span class='notice'>\The [user] starts taking the honeycombs out of \the [src].</span>", "<span class='notice'>You start taking the honeycombs out of \the [src]...</span>")
-		while(honeycombs >= 100 && do_after(user, 30, src))
-			new /obj/item/honey_frame/filled(loc)
-			honeycombs -= 100
-			--frames
-		update_icon()
-		if(honeycombs < 100)
-			to_chat(user, "<span class='notice'>You take all filled honeycombs out.</span>")
+		if(open && !notthebees && (bee_count >= 60) && (REALTIMEOFDAY > time_end_smoked))
+			user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+			notthebees = GM
+			user.visible_message(SPAN_DANGER("\The [user] starts jamming \the [GM]'s face into \the [src]!"), SPAN_DANGER("You start jamming \the [GM]'s face into \the [src]!"))
+			add_fingerprint(user)
+			GM.say("What-is-that? What-is-that? WHAT-IS-IT?!!")
+			GM.forceMove(get_turf(src))
+			GM.set_dir(NORTH)
+			GM.apply_effect(2, STUN)
+			var/turns = 0
+			while(do_after(user, 5 SECONDS, src) && (LAZYLEN(GM.grabbed_by)) && !(QDELETED(src) || QDELETED(user) || QDELETED(G) || QDELETED(GM)) && turns < 4)
+				++turns
+				GM.apply_effect(rand(1,20), PAIN)
+				GM.apply_damage(rand(2,8), BRUTE, BP_HEAD, 0, "bee stings", 0, TRUE)
+				GM.emote("scream")
+				ADJ_STATUS(GM, STAT_JITTER, 10)
+				GM.say("[pick("Oh noo! Not the bees! NOT THE BEEEES!!", "AAAAAAAAAAAGH!", "AGH! THEY'RE IN MY EYES! MY EYES!")]")
+
+			if(turns > 0)
+				GM.add_chemical_effect(CE_TOXIN, 1)
+			if(turns >= 4 && prob(10))
+				var/obj/item/organ/external/h = GM.get_organ(BP_HEAD)
+				if(h)
+					h.disfigure(PAIN) //Set it to PAIN since otherwise it has pre-written flavor text that completely doesn't fit the situation
+					user.visible_message(SPAN_WARNING("\The [GM]'s [h] has swollen beyond recognition!"))
```

Comment:
`SPAN_DANGER` would be more appropriate here I think.

Reply:
True